src/components/RangeSelector.js

```diff
@@ -130,20 +172,30 @@ class RangeSelector extends React.Component {
       newState[this.state.dragging.toLowerCase() + 'Pixels'] = newPixels;
       newState[this.state.dragging.toLowerCase() + 'Value'] = newValue;
 
-      this.setState(newState);
+      this.setState(newState, () => {
+        this.props['on' + this.state.dragging + 'DragStop'](this.state[this.state.dragging.toLowerCase() + 'Value']);
+
+        if (this.state.trackClicked) {
+          this.setState({
+            dragging: false,
+            trackClicked: false
+          });
+        }
+      });
```

Comment:
Played around with this as well, and we can get rid of the `setState` callback method. Not only do we get rid of the callback, but it makes it a little more readable:
```javascript
if (this.state.trackClicked) {
  newState.dragging = false;
  newState.trackClicked = false;
}

this.props['on' + this.state.dragging + 'DragStop'](newState[this.state.dragging.toLowerCase() + 'Value']);

this.setState(newState);
```